src/main/java/org/jabref/logic/importer/fetcher/MathSciNet.java

```diff
@@ -102,34 +115,123 @@ public URL getUrlForIdentifier(String identifier) throws URISyntaxException, Mal
-            BibtexParser bibtexParser = new BibtexParser(preferences, new DummyFileUpdateMonitor());
-
             List<BibEntry> entries = new ArrayList<>();
+
             try {
                 // Depending on the type of query we might get either a json object or directly a json array
                 JsonNode node = new JsonNode(response);
+
                 if (node.isArray()) {
                     JSONArray entriesArray = node.getArray();
                     for (int i = 0; i < entriesArray.length(); i++) {
-                        String bibTexFormat = entriesArray.getJSONObject(i).getString("bib");
-                        entries.addAll(bibtexParser.parseEntries(bibTexFormat));
+                        JSONObject entryObject = entriesArray.getJSONObject(i);
+                        BibEntry bibEntry = jsonItemToBibEntry(entryObject);
+                        entries.add(bibEntry);
                     }
                 } else {
                     var element = node.getObject();
-                    JSONArray entriesArray = element.getJSONObject("all").getJSONArray("results");
-                    for (int i = 0; i < entriesArray.length(); i++) {
-                        String bibTexFormat = entriesArray.getJSONObject(i).getString("bibTexFormat");
-                        entries.addAll(bibtexParser.parseEntries(bibTexFormat));
+
+                    if (element.has("all")) {
+                        JSONArray entriesArray = element.getJSONObject("all").getJSONArray("results");
+                        for (int i = 0; i < entriesArray.length(); i++) {
+                            JSONObject entryObject = entriesArray.getJSONObject(i);
+                            BibEntry bibEntry = jsonItemToBibEntry(entryObject);
+                            entries.add(bibEntry);
+                        }
+                    } else if (element.has("results")) {
+                        JSONArray entriesArray = element.getJSONArray("results");
+                        for (int i = 0; i < entriesArray.length(); i++) {
+                            JSONObject entryObject = entriesArray.getJSONObject(i);
+                            BibEntry bibEntry = jsonItemToBibEntry(entryObject);
+                            entries.add(bibEntry);
+                        }
                     }
                 }
-            } catch (JSONException | TokenMgrException e) {
+            } catch (JSONException | ParseException e) {
                 LOGGER.error("An error occurred while parsing fetched data", e);
                 throw new ParseException("Error when parsing entry", e);
             }
             return entries;
         };
     }
 
+    private BibEntry jsonItemToBibEntry(JSONObject item) throws ParseException {
+        try {
+            BibEntry entry = new BibEntry(StandardEntryType.Article);
+            // Set fields based on the mappings
+            for (Map.Entry<StandardField, List<String>> mapEntry : fieldMappings.entrySet()) {
```

Comment:
Even if I commented below to use the fieldmappings, I would handle author and keyword separatetly. You are explicitly handling these two cases below. Why not

```java
entry.put(StandardField.AUTHOR, toAuthors(item.optJSONArray("authors"));
entry.put(StandardField.KEYWORDS, toAuthors(item.optJSONArray("primaryClass"));
```

and then the foreach for the other field mappings?

Reply:
Good point, will change this.